src/bits.hpp

```diff
@@ -27,61 +27,61 @@
 // 64 * 2^16. 2^17 values, each value can store 64 bits.
 #define kMaxSizeBits 8388608
 
-// A stack vector of length 5, having the functions of std::vector needed for Bits.
-struct SmallVector {
-    typedef uint16_t size_type;
-
-    SmallVector() noexcept { count_ = 0; }
-
-    uint64_t& operator[](const uint16_t index) { return v_[index]; }
+template <class item_type, class size_type_arg, unsigned capacity>
```

Comment:
Here's my argument against making `item_type` a template parameter:

As it stands in your patch, it's not correct. The template parameter cannot be an arbitrary type, since they're all constructed up-front they must be default constructable. In order to make this more correct, you would need something like:
```
static_assert(std::is_integral_t<item_type>, "InlineVector inly supports integral types");
```

Which brings me to my next point. We don't currently need this additional generality/complexity. I don't think it makes sense to make things more general/complex just for the sake of it (in fact, I would argue that keeping everything as specific as possible makes for easier-to-read programs).

Now, you say you'll need this template parameter in your next PR. So, let's punt this change until that PR.

Reply:
> The template parameter cannot be an arbitrary type, since they're all constructed up-front they must be default constructable. 

Yeah, I figured that out already in my next patch. Let's see if you like it ;)
Added static assert for now.